src/java/org/apache/cassandra/db/ReadQuery.java

```diff
@@ -256,4 +256,12 @@ default void maybeValidateIndex()
     default void trackWarnings()
     {
     }
+
+    /**
+     * @return true given read query is a top-k request
```

Comment:
```suggestion
     * @return {@code true} if this is a top-k query
```

Reply:
Maybe we could add in the JavaDoc for this method that top-k queries are necessarily based on the ordering provided by an index. Otherwise, the implementation of this method in `ReadCommand` should be based on the `RowFilter` expressions rather than delegating to the index.